Mention vector and array handling for parquet files (#3151)
neo4j/docs-operations: modules/ROOT/pages/import.adoc

Reviewer: I guess this needs to be added here as well.
```suggestion
For Parquet data, this is only needed if the vector is encoded as a string.
```

Author: Done.

Reviewer: ```suggestion
For CSV data, this value must be different from the one specified in the `--delimiter` option.
For Parquet data, this is only needed if the vector is encoded as a string.
```

|--array-delimiter=<char>
|Delimiter character between array elements within a value in CSV data. Also accepts `TAB` and e.g. `U+20AC` for specifying a character using Unicode.
For CSV data, this value must be different from the one specified in the `--delimiter` option.

> For Parquet data, this is only needed if the array is encoded as a string.

Reviewer: Is this added to the description in the monorepo or just here?

Author: Where exactly do you refer this to be added?

Reviewer: To the description of `--array-delimiter`.

|--vector-delimiter=<char>
|label:new[Introduced in 2025.10] Delimiter character between vector coordinates within a value in CSV data. Also accepts `TAB` and e.g. `U+20AC` for specifying a character using Unicode.
For CSV data, this value must be different from the one specified in the `--delimiter` option.

> For Parquet data, this is only needed if the vector is encoded as a string.

Reviewer: Is this added to the descriptions in the monorepo?

Author: Where exactly do you refer this to be added?

Arrays are not affected by the `--normalize-types` flag.
For example, if you want a byte array to be stored as a Cypher long array, you must explicitly declare the property as `long[]`.

> Array values can also be sourced directly from a `LIST` typed column with Parquet files, in which case the `--array-delimiter` option is not needed.

Reviewer: ```suggestion
Starting with Neo4j 2026.06, array values can also be sourced directly from a `LIST` typed column with Parquet files, in which case the `--array-delimiter` option is not needed.
```

Author: This has been there from the initial calver release, that's why I haven't specified a version.

> Furthermore, values read from `LIST` typed columns are mapped to their corresponding Cypher types, e.g. `LIST<INT>` is mapped to `int[]`, `LIST<STRING>` is mapped to `string[]`, and so on and there is no need to specify the type in the header.

Reviewer: ```suggestion
Furthermore, values read from `LIST` typed columns are mapped to their corresponding Cypher types, e.g., `LIST<INT>` is mapped to `int[]`, `LIST<STRING>` is mapped to `string[]`, and so on, and there is no need to specify the type in the header.
```

----

Note that quotation marks are necessary, since the `,` inside the map would otherwise be interpreted as a column delimiter.

With CSV files and Parquet files where vector values are encoded as strings, by default, vector values are separated by `;`.
A different delimiter can be specified with `--vector-delimiter`.

> Since 2026.06, vector values can also be sourced from a `LIST` typed column with Parquet files, in which case the `--vector-delimiter` option is not needed.

Reviewer: ```suggestion
Starting with Neo4j 2026.06, vector values can also be sourced from a `LIST` typed column with Parquet files, in which case the `--vector-delimiter` option is not needed.
```
